spring-integration-core/src/main/java/org/springframework/integration/gateway/MessagingGatewaySupport.java

```diff
@@ -802,6 +857,13 @@ protected void doStop() {
 		}
 	}
 
+	@Override
+	public void destroy() {
+		this.timers.forEach(MeterFacade::remove);
```

Comment:
Ah! I got it.

Gary, take a look, please, what does that `@SuppressWarnings("deprecation")` do on this class?